### PR TITLE
Merge AllNodeLister with ReadyNodeLister.

### PR DESCRIPTION
**Pull request by krzysztof-jastrzebski**

### CODE REVIEW

File: cluster-autoscaler/utils/kubernetes/listers.go [hunk 1]
```diff
@@ -226,83 +226,59 @@ type NodeLister interface {
...
+func (l *nodeListerImpl) List() ([]*apiv1.Node, error) {
+	nodes, err := l.nodeLister.List(labels.Everything())
 	if err != nil {
 		return []*apiv1.Node{}, err
 	}
-	allNodes := append(make([]*apiv1.Node, 0, len(nodes)), nodes...)
-	return allNodes, nil
+	results := make([]*apiv1.Node, 0, len(nodes))
+	for _, node := range nodes {
+		if l.filter == nil || l.filter(node) {
```
Comment (losipiuk) [inline cluster-autoscaler/utils/kubernetes/listers.go]: move `if l.filter == nil` out of the for loop

Comment (krzysztof-jastrzebski) [reply cluster-autoscaler/utils/kubernetes/listers.go]: Done.

**Pull request merged**